src/test/kotlin/uk/gov/dluhc/notificationsapi/rest/GenerateInviteToRegisterTemplatePreviewIntegrationTest.kt

```diff
@@ -0,0 +1,343 @@
+    }
+
+    @Autowired
+    private lateinit var sourceTypeMapper: SourceTypeMapper
+
+    @Autowired
+    private lateinit var deadlineMapper: DeadlineMapper
+
+    @Autowired
+    private lateinit var languageMapper: LanguageMapper
+
+    @BeforeEach
+    fun setup() {
+        wireMockService.stubCognitoJwtIssuerResponse()
+    }
+
+    @Test
+    fun `should return unauthorized given user with invalid bearer token`() {
+        webTestClient.post()
+            .uri(URI_TEMPLATE)
+            .bearerToken(UNAUTHORIZED_BEARER_TOKEN)
+            .contentType(MediaType.APPLICATION_JSON)
+            .exchange()
+            .expectStatus()
+            .isUnauthorized
+    }
+
+    @Test
+    fun `should return forbidden given no bearer token`() {
+        webTestClient.post()
+            .uri(URI_TEMPLATE)
+            .exchange()
+            .expectStatus()
+            .isForbidden
+    }
+
+    @Test
+    fun `should return not found given non existing template`() {
+        // Given
+        wireMockService.stubNotifyGenerateTemplatePreviewNotFoundResponse(EMAIL_ENGLISH_TEMPLATE_ID)
+        val earliestExpectedTimeStamp = OffsetDateTime.now().truncatedTo(ChronoUnit.MILLIS)
+
+        // When
+        val response = webTestClient.post()
+            .uri(URI_TEMPLATE)
+            .bearerToken(getBearerToken())
+            .contentType(MediaType.APPLICATION_JSON)
+            .withAValidBody(SourceType.POSTAL)
+            .exchange()
+            .expectStatus()
+            .isNotFound
+            .returnResult(ErrorResponse::class.java)
+
+        // Then
+        val actual = response.responseBody.blockFirst()
+        ErrorResponseAssert.assertThat(actual)
+            .hasTimestampNotBefore(earliestExpectedTimeStamp)
+            .hasStatus(404)
+            .hasError("Not Found")
+            .hasMessage("Notification template not found for the given template type")
+            .hasNoValidationErrors()
+    }
+
+    @Test
+    fun `should return bad request given missing request body`() {
+        // Given
+        val earliestExpectedTimeStamp = OffsetDateTime.now().truncatedTo(ChronoUnit.MILLIS)
+
+        // When
+        val response = webTestClient.post()
+            .uri(URI_TEMPLATE)
+            .bearerToken(getBearerToken())
+            .contentType(MediaType.APPLICATION_JSON)
+            .exchange()
+            .expectStatus()
+            .isBadRequest
+            .returnResult(ErrorResponse::class.java)
+
+        // Then
+        val actual = response.responseBody.blockFirst()
+        ErrorResponseAssert.assertThat(actual)
+            .hasTimestampNotBefore(earliestExpectedTimeStamp)
+            .hasStatus(400)
+            .hasError("Bad Request")
+            .hasMessageContaining("Required request body is missing")
+            .hasNoValidationErrors()
+    }
+
+    @Test
+    fun `should return bad request given valid missing request body parameters to gov notify`() {
```

Comment:
This name is a bit confusing to me, is this one better?
```suggestion
    fun `should return bad request given GOV Notify returns bad request due to missing request body parameters`() {
```